src/webview/MessageList.js

```diff
@@ -261,29 +261,32 @@ class MessageListInner extends Component<Props> {
  * "Can", not "will": other conditions can mean we don't want to mark
  * messages as read even when in a narrow where this is true.
  */
-const marksMessagesAsRead = (narrow: Narrow): boolean =>
+const marksMessagesAsRead = (
+  narrow: Narrow,
+  shouldMarkAsReadOnScroll: 'never' | 'always' | 'conversation',
+): 'never' | 'always' | 'conversation' =>
```

Comment:
Except for the search, starred, and mentioned narrows, every other narrows should depend on the currently chosen setting. That's why I added this extra parameter